lab-gurpreet/lib/promise.js

```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+const cowsay = require('cowsay');
+
+module.exports = exports = function(request,response) {
+  return new Promise((resolve, reject) => {
+    let json = '';
+    request.on('data', (data) => {
+      json += data;
+    });
+
+    request.on('end', () => {
+      try{
+        let parsedJson = JSON.parse(json);
+        resolve(parsedJson);
+        response.write(cowsay.say(JSON.parse(json)) + '\n');
```

Comment:
Lines 16 and 17 should be in your server file. Your server should handle the response, not your promise. All your promise should do is handle (parse) your JSON.